### PR TITLE
test: 32-point regression suite for #166 + #167 (scanner/cover + backup/restore)

### PR DESCRIPTION
**Pull request by fabiodalez-dev**
## Regression suite for #166 + #167 — 32 targeted tests

One reusable Playwright test per checklist point across both open PRs (scanner #164/cover #165-#166 and complete backup #162), authored to cover the exact behaviours and edge cases that are hard to trigger by hand (concurrency/locks, maintenance mode, PHP fallback streaming, `.cnf` escaping, failed external cover replacement).

**Base** is `test/combined-166-167` (both PRs merged) because the suite exercises code from both — the diff here is only the suite itself plus a small backup review-fix commit.

### Coverage (32 tests, 8 serial groups)
- **G1 Backup** (1-5): full/db scope, ZIP magic + no `.env`, covers bundled, "Formato legacy" badge
- **G2 Restore** (6-12): DB+file+trigger round-trip, restore-from-file, focus-cancel dialog, upload validation, trust advisory, "Ricarica la pagina", non-JSON error handling
- **G3 Lock** (13-15): concurrent restores rejected, maintenance/lock lifecycle, restore-vs-update lock (deterministic PHP `flock` holder)
- **G4 Permissions** (16-17): staff → 403 on every destructive endpoint, scope selector admin-only
- **G5 Robustness** (18-21, PHP-level): `importViaPhp` streaming round-trip, loan-overlap trigger, `cnfEscape`, pre-update scope plumbing
- **G6 Scanner** (22-25): Enter triggers import not submit, double-Enter guard, "Codice mancante", EAN handling
- **G7 Cover** (26-30): one-step replace (no orphan), URL replace, failed-external keeps old cover (F006), remove+re-pick (F007), scraped+upload no orphan (F002)
- **G8 Cross-cutting** (31-32): i18n keys in all 4 locales, build guard

### How to run
The suite reads `E2E_BASE_URL` / `E2E_APP_ROOT` / `E2E_DB_*` from the environment so it can target any install:

```
E2E_BASE_URL=… E2E_APP_ROOT=… E2E_DB_* … \
  npx playwright test tests/pr166-167-coverage.spec.js --config=tests/playwright.config.js --workers=1
```

Validated 32/32 green on two consecutive runs against a dedicated isolated install (separate port + temp DB). Filesystem assertions target `E2E_APP_ROOT`; restore round-trips self-isolate via the snapshot pattern; rate limits are cleared per login/backup to keep long runs stable.

### Also included
A small follow-up to the backup code addressing review findings: symlink-safe `deleteDirectory`, explicit socket host in `mysqli_connect`, fail-loud restore maintenance flag, and admin-only download/delete buttons. (Same fix is also on the #167 branch.)

## Summary by CodeRabbit

* **Bug Fixes**
  * Ripristini più affidabili con miglior gestione della maintenance e verifica scrittura del flag.
  * Corretto comportamento su socket DB per garantire connessioni tramite socket.
  * Evitata discesa in symlink durante cancellazioni ricorsive.

* **UI/UX Improvements**
  * Pulsanti Download/Elimina raggruppati nell’interfaccia backup per admin.

* **Tests**
  * Test estesi e nuovi end-to-end; gestione password DB escapata nei test; aggiornati helper di test.

### CODE REVIEW

**Comment by fabiodalez-dev** [comment]
@coderabbitai review

**Comment by fabiodalez-dev** [comment]
@coderabbitai review

File: tests/pr166-167-coverage.spec.js [hunk 1]
```diff
@@ -0,0 +1,1335 @@
...
+    await page.waitForSelector('#importIsbn', { timeout: 10000 });
+    const requests = [];
+    page.on('request', r => { if (r.url().includes('/api/scrape/isbn')) requests.push(r.url()); });
+    await page.fill('#importIsbn', '9788842935780');
+    await page.locator('#importIsbn').press('Enter');
+    // Immediately press Enter again — button should be disabled
+    await page.locator('#importIsbn').press('Enter');
+    await page.waitForTimeout(1500);
+    expect(requests.length).toBe(1);
+  });
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply tests/pr166-167-coverage.spec.js:1005]: Addressed in 475117e4 — both listeners now use a named handler with a symmetric page.off() (tests 9, 23, 24). Validated: G2+G6 11/11 green.

**Comment by fabiodalez-dev** [comment]
Incluso nella release stabile **v0.7.19** (contenuto mergiato in `main`). Chiuso come parte del rilascio 0.7.19.
